src/hooks/Wallet/hooks/UserAddress.ts

```diff
@@ -1,8 +1,19 @@
 import { useConnectWallet } from '@web3-onboard/react'
-import { useMemo } from 'react'
+import { readNetwork } from 'constants/networks'
+import { getArcxClient } from 'lib/arcx'
+import { useEffect, useMemo } from 'react'
 
 export function useUserAddress() {
   const [{ wallet }] = useConnectWallet()
   const userAddress = useMemo(() => wallet?.accounts[0]?.address, [wallet])
+
+  useEffect(() => {
+    if (!userAddress) return
+
+    getArcxClient().then(arcx => {
+      arcx?.connectWallet({ account: userAddress, chain: readNetwork.chainId })
+    })
+  }, [userAddress])
```

Comment:
Not sure if this is the best place to put this.

Reply:
I think this probably belongs in a different hook, wdyt?